.github/workflows/docker-pipeline.yml

```diff
@@ -17,7 +17,15 @@ jobs:
 
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
```

Comment:
I wounder if you could swicth over to https://github.com/openclimatefix/uk-pvnet-app/blob/main/.github/workflows/branch_ci.yml and https://github.com/openclimatefix/uk-pvnet-app/blob/main/.github/workflows/main_ci.yml, this should help simplify the github worflow